tools/deno_dir_test.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+# Check deno dir is created properly
+# Usage: deno_dir_test.py [path to deno dir]
+import os
+import sys
+
+def deno_dir_test(deno_dir):
+    assert os.path.isdir(deno_dir)
+    assert os.path.isdir(os.path.join(deno_dir, "deps"))
+    assert os.path.isdir(os.path.join(deno_dir, "gen"))
+
+
+def main(argv):
+    deno_dir_test(argv[1])
```

Comment:
IMO in this file you want to:

pop the current env (and store it)
rmtree(deno_dir)
run deno with no env flag
assert deno_dir doesn't exist
set env flag (to .deno_test)
run deno
assert deno_dir exists etc
rmtree(deno_dir)
reset back env - even in case of failure

Maybe this is more than @ry wanted?